### PR TITLE
[MM-47565]: In product true up for yearly products

### PR DESCRIPTION
**Pull request by emmyni**
#### Summary

When a new user is created and added to a workspace, check to see if the current subscription is a cloud yearly product. If it is a cloud yearly product, then check if the `OriginallyLicensedSeats` for the yearly subscription is less than the current number of users in the workspace (including the newly added user). If the current number of users exceeds the value of `OriginallyLicensedSeats`, then call the `CreateOrUpdateSubscriptionHistoryEvent` function in enterprise/CWS repo to create a new entry in the `subscriptionHistory` table in CWS. 

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-47565

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/user.go [hunk 1]
```diff
@@ -318,9 +318,30 @@ func (a *App) createUserOrGuest(c request.CTX, user *model.User, guest bool) (*m
...
 
+	_, cwsErr := a.SendSubscriptionHistoryEvent(ruser.Id)
+	if cwsErr != nil {
+		c.Logger().Error("Failed to create/update the SubscriptionHistoryEvent", mlog.Err(cwsErr))
+	}
+
 	return ruser, nil
 }
 
+func (a *App) SendSubscriptionHistoryEvent(userID string) (*model.SubscriptionHistory, error) {
```
Comment (nickmisasi) [inline app/user.go:329]: Can this be easily moved to app/cloud.go? I think it fits better there

File: app/user.go [hunk 2]
```diff
@@ -318,9 +318,30 @@ func (a *App) createUserOrGuest(c request.CTX, user *model.User, guest bool) (*m
...
+	license := a.Srv().License()
+
+	// No need to create a Subscription History Event if the license isn't cloud
+	if !license.IsCloud() {
+		return nil, nil
+	}
+
+	// Get user count
+	userCount, err := a.Srv().Store().User().Count(model.UserCountOptions{})
```
Comment (nickmisasi) [inline app/user.go:338]: More just a note but we might have to keep an eye on replication lag here. The user is created before this function is called but the count function grabs from the read replica so it's possible the user count could not include this new user

**Review by nickmisasi — COMMENTED** [review]
Overall LGTM! Thanks @emmyni 

Going to leave as comment so I can give another once over once you've addressed https://github.com/mattermost/enterprise/pull/1305/files#r1032677301 on the enterprise PR

**Review by ConorMacpherson — APPROVED** [review]
LGTM.
Some comments from Nick I agree with, so I'll throw my approval on now and revisit when I see some changes if necessary.

**Comment by yasserfaraazkhan** [comment]
@nickmisasi @emmyni for some reason Server+CWS always fails, so couldn't test this.
would need your help in verifying the 'seat' count once this is up for testing.

**Review by nickmisasi — APPROVED** [review]

**Pull request merged**